Add Lisk Mainnet information

TalhaMaliktz opened the pull request:
### What was the problem?

This PR resolves #89

Tschakki commented:
@shuse2 where exactly? don't see it.

shuse2 commented:
New constants has been added here

https://github.com/LiskHQ/lisk-elements/pull/857/files#diff-bdb2ea94c1e1061b8d186fdefa04b3b5R15

TalhaMaliktz committed:
Add link to Superchain-Safe portal (#115)

Tschakki committed:
Add Governance page (#118)

* Add new page

* Fix link

* Add treasury section

* Add proposal specs

* Update link

* Restructure sections

* Restructure sections

* Add table

* Add tabs

* Update treasury section

* Update treasury section

* Add staking section

* Update staking section

* Update staking section

* Update staking section

* Update staking section

* Restructure section

* Add link

* Add todo

* Apply suggestions from code review

Co-authored-by: AndreasKendziorra <40799768+AndreasKendziorra@users.noreply.github.com>

* Apply feedback

* Apply feedback

* Fix typo

* Remove sentence

* Update voting power section

* Fix typo

* Apply suggestions from code review

Co-authored-by: Muhammad Talha <13951043+TalhaMaliktz@users.noreply.github.com>

* Fix link

* Update docs/about-lisk/governance.mdx

* Update docs/about-lisk/governance.mdx

Co-authored-by: AndreasKendziorra <40799768+AndreasKendziorra@users.noreply.github.com>

* Update introduction

* Apply feedback

* Add link

* Apply feedback

* Apply feedback

* Comment out unready links

* Add delegate sections

* Add link

* Add delegation page

* Update sidebar

* Update sidebar

* Apply feedback

* Fix link

* Rename page

* Fix link

* Update page title

* Update links

* Remove links

* Add info box

* Add abbreviation

* Fix typo

* Apply feedback

* Apply feedback

* Apply suggestions from code review

Co-authored-by: Muhammad Talha <13951043+TalhaMaliktz@users.noreply.github.com>

---------

Co-authored-by: AndreasKendziorra <40799768+AndreasKendziorra@users.noreply.github.com>
Co-authored-by: Muhammad Talha <13951043+TalhaMaliktz@users.noreply.github.com>

TalhaMaliktz committed:
Add Lisk Mainnet information

Add Lisk Mainnet information

Add Lisk Mainnet information

Add Lisk Mainnet information

Add Lisk Mainnet information

Tschakki commented on docs/building-on-lisk/connecting-to-a-wallet.mdx:
```diff
@@ -23,6 +23,46 @@ import TabItem from '@theme/TabItem';
+    6. Tap the `Save` button to save Lisk as a network.
+  </TabItem>
+  <TabItem value="1click" label="1-Click solution">
+    #### Adding Lisk with 1 click
```
The 1 click solutions for Lisk Mainnet and Lisk Sepolia are exactly the same currently. Please double check how the process works after migration in the Portal, for both Sepolia and Mainnet

TalhaMaliktz replied:
The process is the same! A minor step of ensuring the correct network selection has been added to enhance clarity.

Tschakki commented on docs/building-on-lisk/contracts.md:
```diff
@@ -51,8 +96,7 @@ For more information on Predeploys, see [Predeploys' README](https://github.com/
 | [**ERC-4337 Entrypoint**](https://github.com/ethereum-optimism/specs/blob/main/specs/protocol/preinstalls.md#erc-4337-entrypoint)  | [0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789](https://sepolia-blockscout.lisk.com/address/0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789) 
 | [**LegacyMessagePasser**](https://github.com/ethereum-optimism/specs/blob/main/specs/protocol/predeploys.md#legacymessagepasser)  | [0x4200000000000000000000000000000000000000](https://sepolia-blockscout.lisk.com/address/0x4200000000000000000000000000000000000000) |
-| [**DeployerWhitelist**](https://github.com/ethereum-optimism/specs/blob/main/specs/protocol/predeploys.md#deployerwhitelist)       | [0x4200000000000000000000000000000000000002](https://sepolia-blockscout.lisk.com/address/0x4200000000000000000000000000000000000002) |
-| [**LegacyERC20ETH**](https://github.com/ethereum-optimism/specs/blob/main/specs/protocol/predeploys.md#legacyerc20eth)                | [0xDeadDeAddeAddEAddeadDEaDDEAdDeaDDeAD0000](https://sepolia-blockscout.lisk.com/address/0xDeadDeAddeAddEAddeadDEaDDEAdDeaDDeAD0000) |
```
Why is this removed?

TalhaMaliktz replied:
`LegacyERC20ETH` is a depreciated contract! and is not used anymore.

Tschakki commented on docs/building-on-lisk/deploying-smart-contract/with-Foundry.md:
```diff
@@ -22,10 +22,21 @@ keywords: [
+
 # Deploying a smart contract with Foundry
 
 In this guide, we discuss, the basics of the [Foundry](https://book.getfoundry.sh/) development toolchain and will describe how to create and deploy a smart contract with Foundry to the **Lisk Sepolia** testnet.
```
```suggestion
In this guide, we discuss the basics of the [Foundry](https://book.getfoundry.sh/) development toolchain and will describe how to create and deploy a smart contract with Foundry to the **Lisk Sepolia** testnet.
```

Tschakki commented on docs/building-on-lisk/deploying-smart-contract/with-Foundry.md:
````diff
@@ -325,14 +353,28 @@ Contract [src/NFT.sol:NFT] "0x108872F713A27bc22ca1db8CEefCAC8CbeDdF9E5" is alrea
+    ```bash
+    forge verify-contract <CONTRACT_ADDRESS> \
+    ./src/<CONTRACT_FILE>.sol:<CONTRACT_NAME> \
+    --chain 4202 \
````
```suggestion
    --chain 1135 \
```

Tschakki commented on docs/building-on-lisk/interacting-with-the-blockchain/viem.mdx:
```diff
@@ -46,21 +46,42 @@ In order to read data from Lisk, you need to create a **Public client**.
+    }).extend(publicActionsL2()) 
+
+    export const publicClientL1 = createPublicClient({
+      chain: lisk,
```
shouldn't this be ethereum?

TalhaMaliktz replied:
```suggestion
      chain: ethereum,
```

TalhaMaliktz committed:
Apply suggestions from Mona's review

Co-authored-by: Mona Bärenfänger <tschakki@posteo.de>

TalhaMaliktz committed:
Update docs/building-on-lisk/interacting-with-the-blockchain/viem.mdx

TalhaMaliktz committed:
Update connecting-to-a-wallet.mdx

TalhaMaliktz committed:
Merge branch 'main' into 89-add-network-info-for-lisk-mainnet

TalhaMaliktz committed:
Comment out Governance related information

Tschakki reviewed (dismissed):
LGTM 🎉

nagdahimanshu commented on docs/about-lisk/network-info.md:
```diff
@@ -21,6 +21,19 @@ A reference page containing information about all the networks supported by Lisk
 
+### Lisk Mainnet
+
+| Name                    | Value                                                                       |
+| :--------------         | :-------------------------------------------------------------------------- |
+| **Network Name**        | Lisk                                                                        |
+| **HTTP RPC Endpoint**   | [https://rpc.api.lisk.com](https://rpc.api.lisk.com)                        |
+| **WS RPC Endpoint**     | [wss://ws.api.lisk.com](wss://ws.api.lisk.com)                              |
+| **Chain ID**            | 1135                                                                        |
+| **Currency Symbol**     | ETH                                                                         |
+| **Block Explorer**      | https://blockscout.lisk.com                                                 |
+| **Block Explorer API**  | https://blockscout.lisk.com/api                                             |
+| **Lisk Bridge** | https://bridge.lisk.com                                                     |
+
```
Do you think it is better to combine both tables into one, for example something like this:

| Name                   | Lisk Mainnet                                         | Lisk Sepolia Testnet                                                 |
| :--------------------- | :--------------------------------------------------- | -------------------------------------------------------------------- |
| **Network Name**       | Lisk                                                 | Lisk Sepolia Testnet                                                 |
| **HTTP RPC Endpoint**  | [https://rpc.api.lisk.com](https://rpc.api.lisk.com) | [https://rpc.sepolia-api.lisk.com](https://rpc.sepolia-api.lisk.com) |
| **WS RPC Endpoint**    | [wss://ws.api.lisk.com](wss://ws.api.lisk.com)       | [wss://ws.sepolia-api.lisk.com](wss://ws.sepolia-api.lisk.com)       |
| **Chain ID**           | 1135                                                 | 4202                                                                 |
| **Currency Symbol**    | ETH                                                  | ETH                                                                  |
| **Block Explorer**     | https://blockscout.lisk.com                          | https://sepolia-blockscout.lisk.com                                  |
| **Block Explorer API** | https://blockscout.lisk.com/api                      | https://sepolia-blockscout.lisk.com/api                              |
| **Lisk Bridge**        | https://bridge.lisk.com                              | https://sepolia-bridge.lisk.com                                      |
| **Lisk Faucet**        | NA                                                   | https://app.optimism.io/faucet                                       |

TalhaMaliktz replied:
Don't mind it either way but this separation allows linking to either mainnet or testnet info individually from other pages of the documentation which is not possible if both tables are combined.

nagdahimanshu commented on docs/building-on-lisk/add-token-to-lisk/custom-token.mdx:
```diff
@@ -10,26 +10,36 @@ keywords:
 
-In this tutorial you'll learn how to bridge a custom ERC-20 token from Ethereum to Lisk Sepolia using the [Standard Bridge system](https://docs.optimism.io/builders/dapp-developers/bridging/standard-bridge).
-This tutorial is meant for developers who already have an existing ERC-20 token on Ethereum and want to create a bridged representation of that token on Lisk Sepolia.
+In this tutorial you'll learn how to bridge a custom ERC-20 token from Ethereum or Sepolia to Lisk or Lisk Sepolia network, respectivley.
```
```suggestion
In this tutorial, you will learn how to bridge a custom ERC-20 token from Ethereum or Sepolia to the Lisk or Lisk Sepolia network, respectively.
```

nagdahimanshu commented on docs/building-on-lisk/add-token-to-lisk/custom-token.mdx:
```diff
@@ -42,18 +52,18 @@ You can use the [Superchain Faucet](https://app.optimism.io/faucet?utm_source=do
 You will need an L1 ERC-20 token for this tutorial.
-If you already have an L1 ERC-20 token deployed on Sepolia, you can skip this step.
-Otherwise, you can use the testing token located at [`0x5589BB8228C07c4e15558875fAf2B859f678d129`](https://sepolia.etherscan.io/address/0x5589BB8228C07c4e15558875fAf2B859f678d129) that includes a `faucet()` function that can be used to mint tokens.
+If you already have an L1 ERC-20 token, you can skip this step.
```
Just for consistency
```suggestion
If you already have an L1 ERC-20 token deployed on Ethereum Mainnet or Sepolia, you can skip this step.
```

nagdahimanshu commented on docs/building-on-lisk/connecting-to-a-wallet.mdx:
```diff
@@ -23,6 +23,47 @@ import TabItem from '@theme/TabItem';
+    | Name            | Value                                                                       |
+    | :-------------- | :-------------------------------------------------------------------------- |
+    | Network Name    | Lisk                                                                        |
+    | RPC Endpoint    | [	https://rpc.api.lisk.com](https://rpc.api.lisk.com)                       |
```
```suggestion
    | RPC Endpoint    | [ https://rpc.api.lisk.com](https://rpc.api.lisk.com)      |
```

nagdahimanshu commented on docs/governance/delegation.md:
```diff
@@ -0,0 +1,109 @@
+
+Delegates are required to:
+
+- Follow the processes outlined in the Governance documentation of the Lisk DAO
```
Should we link the `Governance documentation` here in case available?

TalhaMaliktz replied:
The governance-related docs are hidden for now and are not available publicly. They will be released publically next week and then the appropriate links will be added.

nagdahimanshu commented on docs/governance/overview.mdx:
```diff
@@ -0,0 +1,179 @@
+:::
+
+### Voting power
+Every account holding **LSK** is able to **stake/lock** a specific amount of LSK tokens to get some voting power  and [staking rewards](#staking-rewards).
```
```suggestion
Every account holding **LSK** is able to **stake/lock** a specific amount of LSK tokens to get some voting power and [staking rewards](#staking-rewards).
```

nagdahimanshu commented on docs/governance/overview.mdx:
```diff
@@ -0,0 +1,179 @@
+
+##### Boosting the voting power
+Users have the option to pause the countdown of the locking period.
+That means, the counting towards the end of the locking period is paused until the user decides to resume it.
```
Optional
```suggestion
This means that the counting towards the end of the locking period is paused until the user decides to resume it.
```

nagdahimanshu commented on docs/governance/overview.mdx:
```diff
@@ -0,0 +1,179 @@
+| **General** |  General proposals about protocol parameters or the project direction in general that may be followed by the Onchain Foundation or not, depending also on feasibility and cost.  | No  |
+
+#### Proposal evaluation
+The following requirements must be fulfilled for a proposal to pass:
```
```suggestion
The following requirements must be fulfilled for a proposal to approve:
```

nagdahimanshu commented on docs/governance/overview.mdx:
```diff
@@ -0,0 +1,179 @@
+
+#### Implementation of funding proposals
+The transfer from the [Lisk DAO treasury](#treasury) can be executed by anyone, once a funding proposal is approved.
+Funding proposals are **binding**, which means that no one, not even the Onchain Foundation, can prevent the transfer if the proposal passes.
```
```suggestion
Funding proposals are **binding**, which means that no one, not even the Onchain Foundation, can prevent the transfer if the proposal approves.
```

nagdahimanshu commented on docs/governance/overview.mdx:
```diff
@@ -0,0 +1,179 @@
+If it is, the Foundation will act diligently and in a commercially reasonable manner to cause the proposal to be implemented.
+
+### Administrative role of the Onchain Foundation
+The foundation will have an administrative role and will take care of the following
```
```suggestion
The foundation will have an administrative role and will take care of the following:
```

nagdahimanshu commented on docs/lisk-tools/utilities.md:
```diff
@@ -31,6 +31,7 @@ Web3 Functions serve as a comprehensive tool, enabling developers to effortlessl
 ### Safe 
 [Safe](https://docs.safe.global) is the account abstraction leader on Ethereum and the EVM with the most secure smart wallet infrastructure and platform.
 Safe brings digital ownership of accounts to everyone by building universal and open contract standards for the custody of digital assets, data, and identity.
+To create a Safe account with Lisk, [connect your wallet](https://safe.optimism.io/welcome?chain=lisksep) through the Superchain-Safe portal.
```
May be we can update it to connect to Lisk mainnet:
```suggestion
To create a Safe account with Lisk, [connect your wallet](https://safe.optimism.io/welcome?chain=lisk) through the Superchain-Safe portal.
```

nagdahimanshu commented on docs/lisk-tools/bridges.mdx:
```diff
@@ -0,0 +1,90 @@
+
+
+:::info
+After initiating your withdrawal, a 7-day [challenge period](https://docs.optimism.io/builders/dapp-developers/bridging/messaging#for-l2-to-l1-transactions) needs to be observed as a security measure. Once the challenge period has passed the withdrawal can be completed with a final transaction.
```
```suggestion
After initiating your withdrawal, a 7-day [challenge period](https://docs.optimism.io/builders/dapp-developers/bridging/messaging#for-l2-to-l1-transactions) needs to be observed as a security measure. Once the challenge period has passed, the withdrawal can be completed with a final transaction.
```

nagdahimanshu reviewed (changes requested):
Looks good overall, left minor comments.

TalhaMaliktz committed:
Apply suggestions from Himanshu's review

Co-authored-by: Himanshu Nagda <69150921+nagdahimanshu@users.noreply.github.com>

TalhaMaliktz committed:
Apply Himanshu's feedback

nagdahimanshu reviewed (approved):
LGTM 👍
